src/transformers/models/whisper/processing_whisper.py

```diff
@@ -91,3 +96,22 @@ def decode(self, *args, **kwargs):
         the docstring of this method for more information.
         """
         return self.tokenizer.decode(*args, **kwargs)
+
+    def get_prompt_ids(self, text: str):
+        """Converts prompt text to IDs that can be passed to [`~WhisperForConditionalGeneration.generate`]."""
+        _verify_no_special_tokens(text, self.tokenizer.all_special_tokens)
+        prompt_token = "<|startofprev|>"
+        batch_encoding = self.tokenizer(prompt_token, text.strip(), add_prefix_space=True, add_special_tokens=False)
+        batch_encoding.convert_to_tensors(tensor_type="pt")
+        return batch_encoding["input_ids"]
+
+
+def _verify_no_special_tokens(text: str, all_special_tokens: List[str]):
```

Comment:
Agree with @hollance that checking for any tokens > 50364 is probably cleaner and faster here if we can manage it https://github.com/huggingface/transformers/pull/22496#discussion_r1185251784

Reply:
Do you mean 50364 hard coded or the min of the special ids? I'm thinking it's the latter since token-id mappings can vary between tokenizers and some special ids are < 50364.

Reply:
This could crash if someone use custom vocab when hardcoding the index.

Reply:
Yeah best not hardcode it but use `all_special_ids[-1] + 1` which is the first timestamp token.

Reply:
Also per @versae 's linked issue above timestamps aren't properly tokenized right now so won't be caught by this check

Reply:
Couldn't we just add the 1500 timestamps tokens to the tokenizer? That would be in sync with the existing OpenAI implementation. Or maybe add them to the `WhisperTokenizer` class as special `AddedToken`'s? It would mean changing statements like ` all_special_ids[-1] + 1` into  `tokenizer.convert_tokens_to_ids("<|0.00|>")`, but would also enable timestamps encoding out-of-the-box without impacting decoding.

A bit off-topic, but changing the tokenizer to encode the timestamps tokens so training can take those into account is kinda hard, as it touches the code in tons of places (think the trie that is built internally).